ee/src/main/java/org/jboss/as/ee/component/ManagedReferenceMethodInjectionInterceptorFactory.java

```diff
@@ -105,7 +105,7 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
                 final InvocationType invocationType = context.getPrivateData(InvocationType.class);
                 try {
                     context.putPrivateData(InvocationType.class, InvocationType.DEPENDENCY_INJECTION);
-                    method.invoke(target, reference.getInstance());
+                    method.invoke(target, reference != null ? reference.getInstance() : null);
```

Comment:
This and the other change in this class are questionable.  There's an 'optional' field and logic at L99 to deal with cases where this is null and optional==true. And then later on the code will fail if this is null and optional==false.

It's not clear the thing to do here is to no longer fail. More intuitive would be to fail earlier on in a clearer way than an NPE.

Reply:
@bstansberry I have found a existing logic/exception for "field" [https://github.com/wildfly/wildfly/blob/main/ee/src/main/java/org/jboss/as/ee/logging/EeLogger.java#L1043](https://github.com/wildfly/wildfly/blob/main/ee/src/main/java/org/jboss/as/ee/logging/EeLogger.java#L1043). 
Overloading EeLogger

```java
    @Message(id = 131, value = "ManagedReference was null and injection is not optional for injection into method %s")
    RuntimeException managedReferenceWasNull(Method method);
```
Adding after L101:

```java
            if (reference == null) {
                throw EeLogger.ROOT_LOGGER.managedReferenceWasNull(method);
            }
```

Does this solution address your concerns?